diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FunctonASTNodeUtils.kt

```diff
@@ -1,11 +1,36 @@
 package org.cqfn.diktat.ruleset.utils
 
 import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.IDENTIFIER
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import org.cqfn.diktat.ruleset.rules.PackageNaming.Companion.PACKAGE_PATH_ANCHOR
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.kdoc.parser.KDocKnownTag
 import org.jetbrains.kotlin.psi.KtParameter
 import org.jetbrains.kotlin.psi.KtParameterList
 
+/**
+ * Checks whether function from this [ElementType.FUN] node is from test class based on annotations and file location
+ */
+fun ASTNode.isTestFun(filePathParts: List<String>): Boolean {
```

Comment:
could you please split this check for two checks? 
Test annotation and test path.

It will be useful for complex checks. For example that functions with Test annotation are placed in files with "Test" name/in test directories